Modules/Scripted/VPAWVisualize/VPAWVisualize.py

```diff
@@ -322,24 +284,44 @@ def onApplyButton(self):
         ):
 
             # Compute output
-            self.logic.process(
-                self.ui.inputSelector.currentNode(),
-                self.ui.outputSelector.currentNode(),
-                self.ui.imageThresholdSliderWidget.value,
-                self.ui.invertOutputCheckBox.checked,
+            list_of_files = self.logic.process(
+                self.ui.DataDirectory.currentPath, self.ui.PatientPrefix.text
             )
-
-            # Compute inverted output (if needed)
-            if self.ui.invertedOutputSelector.currentNode():
-                # If additional output volume is selected then result with inverted
-                # threshold is written there
-                self.logic.process(
-                    self.ui.inputSelector.currentNode(),
-                    self.ui.invertedOutputSelector.currentNode(),
-                    self.ui.imageThresholdSliderWidget.value,
-                    not self.ui.invertOutputCheckBox.checked,
-                    showResult=False,
-                )
+            # Display output
+            self.updateOutputWidgets(list_of_files)
+
+    def clearOutputWidgets(self):
+        outputsFormLayout = self.ui.outputsCollapsibleButton.layout()
+        # Clear away the current output widgets
+        for row_number in reversed(range(outputsFormLayout.rowCount())):
+            # !!! Do we need to delete (recursively) the widgets themselves before we
+            # !!! delete the row from the layout?  Or, is Python garbage collection
+            # !!! sufficient?
+            outputsFormLayout.removeRow(row_number)
+
+    def updateOutputWidgets(self, list_of_files):
+        self.clearOutputWidgets()
+
+        outputsFormLayout = self.ui.outputsCollapsibleButton.layout()
+        for file in list_of_files:
+            try:
+                # Collapsible contains Layout contains Label contains Pixmap
+                fileCollapsible = ctk.ctkCollapsibleButton()
+                fileCollapsible.text = file
+                fileCollapsible.collapsed = True
+                fileLayout = qt.QFormLayout(fileCollapsible)
+                fileLabel = qt.QLabel()
+                fileLabel.setText(file)
+                filePixmap = qt.QPixmap()
+                if not filePixmap.load(file):
+                    raise RuntimeError(f"Could not load pixmap from file {repr(file)}")
```

Comment:
I am getting this
```txt
RuntimeError: Could not load pixmap from file '/home/ebrahim/vpaw-model-workspace/some_other_root_dir/images/1000_INPUT.nrrd'
```
when I try out VPAWVisualize.

I'm not clear on what is the role of `filePixmap` here. Definitely I shouldn't give it a nrrd file, but what is it expecting?

Reply:
It is expecting an image file; this code needs to support any image file types that can be produced by `pediatric_airway_atlas`.  Apparently `QPixmap` does not support `nrrd` images, at least not directly.  I'll figure out what we have to do to make that work.

Reply:
It's a 3D image -- I think you'd just want to load it into a mrml node:
https://slicer.readthedocs.io/en/latest/developer_guide/script_repository.html#load-volume-from-file

Reply:
Now we are supporting volumes, segmentations, ... rather than 2D images.